### PR TITLE
[cdsearch-601] ensure reasonable error handling if no contact with server

### PR DESCRIPTION
**Pull request by baob**

### CODE REVIEW

File: spec/lib/bandiera/client_spec.rb [hunk 1]
```diff
@@ -4,7 +4,41 @@
   let(:base_uri)  { 'http://bandiera.com' }
   let(:api_uri)   { "#{base_uri}/api" }
   let(:logger)    { double.as_null_object }
-  subject         { Bandiera::Client.new(api_uri, logger) }
+  let(:client_instance) { Bandiera::Client.new(api_uri, logger) }
```
Comment (dazoakley) [inline spec/lib/bandiera/client_spec.rb]: Not really sure on this change - `subject` is the standard name for the test target...

File: spec/lib/bandiera/client_spec.rb [hunk 2]
```diff
@@ -32,19 +66,20 @@
...
+      client_instance.enabled?('foo', 'bar', params, options)
     end
   end
 
   describe '#get_feature' do
     let(:group)   { 'pubserv' }
     let(:feature) { 'log-stats' }
     let(:url)     { "#{api_uri}/v2/groups/#{group}/features/#{feature}" }
+    let(:response) { client_instance.get_feature(group, feature) }
+    let(:expected_error_response) {false}
```
Comment (dazoakley) [inline spec/lib/bandiera/client_spec.rb]: :+1:

**Comment by dazoakley** [comment]
Thanks for the update @baob!

Looks like there is an issue with the latest version of rack and older rubies (check travis) - maybe tie to an older version of rack for now?

Other than the 2 comments, looks good to me.

File: spec/lib/bandiera/client_spec.rb (hunk 1 above)
Comment (baob) [inline spec/lib/bandiera/client_spec.rb]: Actually, the way the spec refactoring turned out, I didn't need it.  I'll go back to subject.

**Comment by baob** [comment]
@dazoakley You're right on rack ... within the last couple of weeks they went to rack 2.0.0 with constraints on rubies, we hit the problem on macmillan-utils last week. If it turns out like that one, we may need to  restrain the rack version and drop support for ruby 1.9 too, which should be no big deal.

**Comment by dazoakley** [comment]
Our aim with the support of different versions of Ruby were to track the "officially" supported ones and just make sure we document when the support for an older version is dropped.  According to the [ruby-lang](https://www.ruby-lang.org/en/downloads/) site, we would be safe to jump up to 2.1+. 🎉

File: bandiera-client.gemspec [hunk 3]
```diff
@@ -23,6 +23,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rspec'
   spec.add_development_dependency 'webmock'
   spec.add_development_dependency 'rack-test'
+  spec.add_development_dependency 'rack', '< 2.0.0' # NOTE for the time being, don't force us to use ruby 2.2.2 please.
```
Comment (dazoakley) [inline bandiera-client.gemspec]: 👍

**Pull request merged**